demos/common/cpp/models/src/segmentation_model.cpp

```diff
@@ -49,14 +51,11 @@ void SegmentationModel::prepareInputsOutputs(InferenceEngine::CNNNetwork& cnnNet
 
     outputsNames.push_back(outputsDataMap.begin()->first);
     Data& data = *outputsDataMap.begin()->second;
-    // if the model performs ArgMax, its output type can be I32 but for models that return heatmaps for each
-    // class the output is usually FP32. Reset the precision to avoid handling different types with switch in
-    // postprocessing
-    data.setPrecision(Precision::FP32);
+
     const SizeVector& outSizeVector = data.getTensorDesc().getDims();
     switch (outSizeVector.size()) {
     case 3:
-        outChannels = 0;
+        outChannels = (int)(outSizeVector[0]);
```

Comment:
That is not correct. If model' output is 3D tensor it has shape `B,H,W`, so the `outSizeVector[0]` stands for batch dimension.  It would be more correctly to set it to 1:
```suggestion
        outChannels = 1;
```

Reply:
BTW, at runtime tensor descriptor show CHW as a layout for 3D tensor (in case of deeplabv3 model). Is it Inference Engine issue?

Reply:
May be
@eaidova could you help us?

Reply:
It is OV limitation. We cannot rely on OV tensor description. t is always report default depends on number of dims. 
There is not any information stored in the model which really store its real layout (majority of frameworks does not contains this info and IR is also do not store it), so it is impossible to somehow correctly get real layout

Reply:
confirmed with benchmark_app, in case of batch_size = 2, blob is still reported as CHW layout, but has dimensions (2, 513, 513)